tests/test_lsst.py

```diff
@@ -0,0 +1,928 @@
+from __future__ import with_statement
+import unittest
+import numpy as np
+import warnings
+import os
+import galsim
+from galsim.lsst import LsstCamera, LsstWCS, _nativeLonLatFromRaDec
+from galsim.celestial import CelestialCoord
+import lsst.afw.geom as afwGeom
+
+def haversine(long1, lat1, long2, lat2):
+    """
+    Return the angular distance between two points in radians
+
+    inputs
+    ------------
+    long1 is the longitude of point 1 in radians
+
+    lat1 is the latitude of point 1 in radians
+
+    long2 is the longitude of point 2 in radians
+
+    lat2 is the latitude of point 2 in radians
+
+    outputs
+    ------------
+    the angular separation between points 1 and 2 in radians
+
+    From http://en.wikipedia.org/wiki/Haversine_formula
+    """
+    t1 = np.sin(lat2/2.-lat1/2.)**2
+    t2 = np.cos(lat1)*np.cos(lat2)*np.sin(long2/2. - long1/2.)**2
+    return 2*np.arcsin(np.sqrt(t1 + t2))
+
+
+class NativeLonLatTest(unittest.TestCase):
+
+    def testNativeLonLat(self):
+        """
+        Test that nativeLonLatFromRaDec works by considering stars and pointings
+        at intuitive locations
+        """
+
+        raList = [0.0, 0.0, 0.0, 1.5*np.pi]
+        decList = [0.5*np.pi, 0.5*np.pi, 0.0, 0.0]
+
+        raPointList = [0.0, 1.5*np.pi, 1.5*np.pi, 0.0]
+        decPointList = [0.0, 0.0,0.0, 0.0]
+
+        lonControlList = [np.pi, np.pi, 0.5*np.pi, 1.5*np.pi]
+        latControlList = [0.0, 0.0, 0.0, 0.0]
+
+        for rr, dd, rp, dp, lonc, latc in \
+        zip(raList, decList, raPointList, decPointList, lonControlList, latControlList):
+            lon, lat = _nativeLonLatFromRaDec(rr, dd, rp, dp)
+            self.assertAlmostEqual(lon, lonc, 10)
+            self.assertAlmostEqual(lat, latc, 10)
+
+
+    def testNativeLongLatComplicated(self):
+        """
+        Test that nativeLongLatFromRaDec works by considering stars and pointings
+        at non-intuitive locations.
+        """
+
+        np.random.seed(42)
```

Comment:
This is probably not a huge problem here, but the `numpy.random` PRNG uses a global state variable, so it's not really thread-safe.  I'm guessing these tests aren't in danger of failing for particularly unlucky choices of random numbers, but to be safe you'd be better off using a well-defined state variable:
```
rng = np.random.RandomState(42)
raPointingList = rng.random_sample(nPointings)*2.0*np.pi
decPointingList = rng.random_sample(nPointings)*np.pi - 0.5*np.pi
```